Update screenshots to new colour scheme

jkmassel opened the pull request:
**To Test**

0. Check out this branch. Ensure that all the tests pass (this will tell us if the changes here are compatible with the existing UI tests).
1. Ensure you've installed the complete Noto Sans font family on your local machine: https://fonts.google.com/specimen/Noto+Sans.
2. Generate a set of Android screenshots using `bundle exec fastlane android screenshots`. This will take about an hour and a half. The content should all load instantly, using the mocked responses.
3. Ensure that the `fastlane/metadata` directory contains the new screenshots.
4. Run `bundle exec fastlane android create_promo_screenshots`. It'll generate localized screenshots into the fastlane/promo_screenshots directory.
5. The screenshots should match those currently used as experiments in the Google Play Store.

loremattei commented:
Hey! I'm moving this to 12.7 since 12.6 has been cut. If you want it to land 12.6, please feel free to move it back, target the release branch and ping me to build a new beta.

nbradbury commented:
Just a friendly reminder to please make sure to manually merge `develop` into this branch before you merge this PR, as `develop` now contains the AndroidX migration. The automatic merge can succeed but still break the build. Better to find out now before it breaks `develop`. Thanks!

loremattei commented:
Hey! I'm moving this to 12.8 since 12.7 has been cut. If this PR has to land 12.7, please feel free to move it back, target the release branch and ping me to build a new beta.

jkmassel commented:
@jtreanor – this is working correctly now, except for the fact that it only works with mocking bypassed. 

Do you want to resolve that in this PR, or separately later? I'm ok either way, but I'd want to be careful not to break the existing UI tests by using new server-provided mocked data.

jtreanor commented:
@jkmassel I think we should resolve this now. It should be relatively straightforward to update the mocks for the changes.

How can I help?

jtreanor commented on libs/mocks/WordPressMocks/src/main/assets/mocks/mappings/format-json.php:
```diff
@@ -0,0 +1,82 @@
+<?php
```
I'm confused. Why is there PHP here? 😄

jkmassel commented:
Not ready for review yet! 😅 This was just a WIP commit because I realized I'd be really sad if my laptop died. I've updated the labels accordingly.

It's not actually an _entirely_ different set, but because it's a different site, I split them up, because the number of screenshot mocks will be far higher than the regular mocks, and I don't want to make it hard for folks working on tests to find what they're looking for.

The reason there are so many screenshot mocks is that it's far more complex than the testing ones. We'll need 22 localized versions of the following:
- Notifications (a lot of the text for them is localized server-side)
- Individual Posts (we'll be getting the post content translated)
- ~one third of the stats endpoints that reference posts by name (so that the localized post names show up on stats screenshots)
- Eventually, customized stats endpoints that reference countries of origin correctly
- Site settings (we need the right site name to be shown on the screenshots)
- Reader (we're not doing screenshots of that at the moment, so it can wait)

For this reason I've split them up, and just adjusted the `/sites` endpoint to include this new one. 

And the PHP file is just for properly formatting the `.json` files when I'm making big changes. I'll get rid of it for the final PR 😜

theck13 commented:
We may want to wait until https://github.com/wordpress-mobile/WordPress-Android/issues/10165 is closed and the associated pull request is merged to have the second version of the colors changes included in the screenshots.

jkmassel commented:
@jtreanor – this is _finally_ ready for review!

jtreanor commented:
There are some conflicts to resolve 🙈 

Lets chat some more about the mocks here. I'm curious if we could avoid some of the duplication.

jkmassel commented:
Yeah! The conflicts are mostly related to the release toolkit changes in https://github.com/wordpress-mobile/release-toolkit/pull/49, which is sort of being tested alongside this PR - I figure we could leave those for the moment if you're ok with it, but LMK? :)

jkmassel commented:
@jtreanor – mocks are merged! Everything now lives in the `wpcom` directory.

Let me know your thoughts!

jkmassel commented:
@loremattei – this is ready for review.

@rachelmcr – tagged you in this because it changes the mocks quite dramatically in order to merge in everything needed for screenshots.

Next step after this is merged will be to add it to the mocks repo and ensure it's all compatible with iOS.

rachelmcr commented:
> Next step after this is merged will be to add it to the mocks repo and ensure it's all compatible with iOS.

What do you think of pushing the subtree to the mocks repo from this branch (rather than after this is merged) so the changes can be tested simultaneously on iOS?

rachelmcr commented on libs/mocks/WordPressMocks/src/main/assets/mocks/mappings/wpcom/stats/stats_top-authors-day.json:
```diff
@@ -0,0 +1,99 @@
+        "urlPath": "/rest/v1.1/sites/106707880/stats/top-authors/",
+        "queryParameters": {
+            "period": {
+                "equalTo": "month"
```
This should be `"equalTo": "day"` instead of `"equalTo": "month"`, right? I'm getting a mocks failure because `/rest/v1.1/sites/106707880/stats/top-authors/?period=day&max=7&date=2020-01-21&locale=en_US` doesn't have a match.

jkmassel replied:
Yep! Fixed

rachelmcr commented on libs/mocks/WordPressMocks/src/main/assets/mocks/mappings/wpcom/stats/stats_search-terms_day.json:
```diff
@@ -0,0 +1,35 @@
+                "matches": "[0-9]+"
+            },
+            "date": {
+                "equalTo": "2019-07-17"
```
I believe this needs to be `"matches": "(.*)"` as with our other stats requests. Otherwise, I get mocks failures because the request (e.g. `/rest/v1.1/sites/106707880/stats/search-terms/?period=day&max=7&date=2020-01-21&locale=en_US`) doesn't have any matches.

jkmassel replied:
For sure – this is fixed!

rachelmcr commented on libs/mocks/WordPressMocks/src/main/assets/mocks/mappings/wpcom/stats/stats.json:
```diff
@@ -1,7 +1,15 @@
     "request": {
         "method": "GET",
-        "urlPattern": "/rest/v1.1/sites/158396482/stats(/+)?($|\\?.*)"
+        "urlPath": "/rest/v1.1/sites/106707880/stats//",
```
There's an extra `/` at the end of the path here, causing a mocks failure.

jkmassel replied:
I was looking at this one trying to remember what happened, because I thought there was a good reason – turned out there was – iOS requires the `//`, or it fails to match the request.

IMHO – this is similar to the issue with the random image IDs and we should leave it with a single slash as it:

A) Doesn't break the tests on iOS (whereas it _does_ break things on Android)
B) Is a bug in WPiOS that should probably be fixed

I tried using the urlPattern instead, but despite trying several different ways, WireMock is refusing to match that URL for whatever reason :(

jkmassel replied:
I figured it out – over-using `urlPattern` in stats broke this, so I've replaced all instances of it there with this, and it now works correctly and fixes this issue.

jkmassel replied:
(FWIW I still think we should fix the bug in iOS that causes this behaviour)

rachelmcr replied:
> iOS requires the //, or it fails to match the request.

o.O Agreed that the bug should be fixed. Could you open an issue on the iOS side for that bug?
